app/src/main/java/com/pose/move/data/preference/AuthTokenProvider.kt

```diff
@@ -0,0 +1,6 @@
+package com.pose.move.data.preference
+
+interface AuthTokenProvider {
+
+    fun getToken(): String
```

Comment:
I'd suggest making it suspend. It is usually blocking i/o call.
Since you have only 1 function, you could convert this into a SAM (not necessary, only if you wish to do so)